### PR TITLE
Revert "[Core] Add retry exception allowlist for user-defined filteri…

### PR DESCRIPTION
**Pull request by SongGuyang**
The related issue https://github.com/ray-project/ray/issues/26287

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

**Review by clarkzinzow — APPROVED** [review]
Yep we should revert and then open a PR unreverting that fixes the Windows test. I'll take a look at doing the latter tomorrow.

**Comment by SongGuyang** [comment]
@edoakes @ericl @maxpumperla @pcmoritz @richardliaw @simon-mo need other code owner approval, who can click?

**Review by stephanie-wang — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Review by fishbone — APPROVED** [review]

**Review by wuisawesome — APPROVED** [review]

**Review by ericl — APPROVED** [review]

**Pull request merged**
